internal/generator/backend/template/zkpschemes/groth16/groth16.prove.go.tmpl

```diff
@@ -216,9 +216,8 @@ func Prove(r1cs *cs.R1CS, pk *ProvingKey, fullWitness witness.Witness, opts ...b
 
 		var krs, krs2, p1 curve.G1Jac
 		chKrs2Done := make(chan error, 1)
-		sizeH := int(pk.Domain.Cardinality - 1) // comes from the fact the deg(H)=(n-1)+(n-1)-n=n-2
 		go func() {
-			_, err := krs2.MultiExp(pk.G1.Z, h[:sizeH], ecc.MultiExpConfig{NbTasks: n / 2})
+			_, err := krs2.MultiExp(pk.G1.Z, h, ecc.MultiExpConfig{NbTasks: n / 2})
```

Comment:
I see we now use the full h here. Are we safe to assume that any indices beyond the n-2 degree are 0?

Reply:
I saw the new computeH, all good here. Maybe we can add a comment or something though, referring to how Circom does it in the same way